addons/sale/models/sale.py

```diff
@@ -289,6 +292,26 @@ def copy_data(self, default=None):
             default['order_line'] = [(0, 0, line.copy_data()[0]) for line in self.order_line.filtered(lambda l: not l.is_downpayment)]
         return super(SaleOrder, self).copy_data(default)
 
+    @api.multi
+    def name_get(self):
+        res = []
+        for order in self:
+            name = order.name
+            if order.partner_name:
+                name = '%s - %s' % (name, order.partner_name)
+            res.append((order.id, name))
+        return res
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
```

Comment:
Nope. Please do not do things like that. It makes no sense. You can use a filter_domain in the search view if you want.